src/client.ts

```diff
@@ -7410,7 +7410,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         return this.http.authedRequest(Method.Post, path, undefined, undefined, { prefix: "" });
     }
 
-    private async fetchClientWellKnown(): Promise<void> {
+    public async fetchClientWellKnown(): Promise<void> {
```

Comment:
How about `protected`?  That would avoid altering the public API of `MatrixClient`.
```suggestion
    protected async fetchClientWellKnown(): Promise<void> {
```